### PR TITLE
Support pandas>=0.18.1

### PR DESCRIPTION
**Pull request by yankees714**
For [compatibility with zipline](https://github.com/quantopian/zipline/blob/fbdfaf82410e6e5a9c9e7e4daa738599a6679683/setup.py#L170).

### CODE REVIEW

**Comment by twiecki** [comment]
We should probably add this to travis as well.

**Comment by yankees714** [comment]
@twiecki how would I go about doing that?

**Comment by twiecki** [comment]
See e.g. https://github.com/quantopian/empyrical/blob/master/.travis.yml#L4

**Comment by yankees714** [comment]
Ah got it, thanks. Do we want all python/pandas combinations?

**Comment by twiecki** [comment]
Yes.

**Comment by twiecki** [comment]
Need to wait for https://github.com/quantopian/pyfolio/pull/442 to fix the current breakage.

**Comment by vikram-narayan** [comment]
merging #442 once tests pass

**Comment by twiecki** [comment]
@yankees714 Can you rebase from master?

File: .travis.yml [hunk 1]
```diff
@@ -1,10 +1,26 @@
 language: python
 sudo: false
 
-python:
-  - "2.7"
-  - "3.5"
-  - "3.6"
+matrix:
```
Comment (vikram-narayan) [inline .travis.yml]: is adding time to the travis build something we're worried about?

File: pyfolio/risk.py [hunk 2]
```diff
@@ -432,9 +432,22 @@ def compute_volume_exposures(shares_held, volumes, percentile):
...
     grossed_frac = shares_grossed.divide(volumes)
 
-    longed_threshold = 100*longed_frac.quantile(percentile, axis='columns')
-    shorted_threshold = 100*shorted_frac.quantile(percentile, axis='columns')
-    grossed_threshold = 100*grossed_frac.quantile(percentile, axis='columns')
+    # NOTE: To work around a bug with nan-handling in pandas 0.18,
+    #       drop all-nan rows before the call to `quantile`, and then
+    #       restore them by reindexing the final result. This is fixed
+    #       in pandas 0.19.
+    longed_threshold = 100*longed_frac.dropna(how='all').quantile(
```
Comment (vikram-narayan) [inline pyfolio/risk.py]: will this fail flake8 if there aren't spaces around `*`?

Comment (yankees714) [reply pyfolio/risk.py]: Apparently not, as I was just leaving this as it was. For some reason thought we were consistently not including the whitespace in this project, but now that I look again it seems like we do include it more often that not. I'll update this and the other occurrences in this file.

File: .travis.yml (hunk 1 above)
Comment (yankees714) [reply .travis.yml]: Possibly, any thoughts on this @twiecki, @richafrank?

Comment (twiecki) [reply .travis.yml]: I'm not worried about it.

Comment (richafrank) [reply .travis.yml]: Each job in the matrix would still take the same amount of time, so we wouldn't suffer any new timeouts. I definitely think we should test an environment that matches what we test for zipline; the others I'm less concerned about. I think we have 4 workers - What's the wall time for a build now?

Is it possible for us to specify the factor, so travis [expands](https://docs.travis-ci.com/user/customizing-the-build#Build-Matrix) the matrix itself?

Comment (yankees714) [reply .travis.yml]: For the last build on this branch it was 20-30 minutes per job, a bit over an hour total.

Comment (yankees714) [reply .travis.yml]: Also yeah looks like we should be able to have travis do the expansion, I'll take a look at that

Comment (richafrank) [reply .travis.yml]: An hour sounds kinda painful to me, but as consumers, what do you all think?

**Comment by yankees714** [comment]
@vikram-narayan I realized there was an issue with one of the workarounds, so I had to modify it (a391f8a), mind taking another look?

File: pyfolio/risk.py [hunk 3]
```diff
@@ -432,22 +435,26 @@ def compute_volume_exposures(shares_held, volumes, percentile):
...
-    #       drop all-nan rows before the call to `quantile`, and then
-    #       restore them by reindexing the final result. This is fixed
-    #       in pandas 0.19.
-    longed_threshold = 100 * longed_frac.dropna(how='all').quantile(
-        percentile,
+    # NOTE: To work around a bug in `quantile` with nan-handling in
+    #       pandas 0.18, use np.nanpercentile by applying to each row of
+    #       the dataframe. This is fixed in pandas 0.19.
+    #
+    # longed_threshold = 100*longed_frac.quantile(percentile, axis='columns')
```
Comment (vikram-narayan) [inline pyfolio/risk.py]: cruft?

**Comment by vikram-narayan** [comment]
actually looks like the travis build failed, is that due to the most recent change?

**Comment by twiecki** [comment]
@vikram-narayan I think conda is still broken.

**Comment by vikram-narayan** [comment]
I think @yankees714 included a workaround for that, it looks like python 3.6 and pandas 0.18.1 are incompatible, from one of the travis builds:
```
UnsatisfiableError: The following specifications were found to be in conflict:
  - pandas 0.18.1* -> numpy 1.10* -> python 2.7* -> openssl 1.0.1*
  - python 3.6*
Use "conda info <package>" to see the dependencies for each package.
```

File: pyfolio/risk.py (hunk 3 above)
Comment (yankees714) [reply pyfolio/risk.py]: I was just leaving these as a reference for when we no longer need to support 0.18, does that sound alright?

Comment (vikram-narayan) [reply pyfolio/risk.py]: yup sounds good

**Comment by yankees714** [comment]
Only remaining concern here was build time - it currently stands at 45 minutes which seems tolerable to me. Going to merge, we can cut time later if need be.

**Comment by twiecki** [comment]
Awesome! Should probably release a new version then.

**Comment by joshpayne** [comment]
@twiecki should we align to completion of the first round of perf attribution studies/tearsheets?

**Comment by twiecki** [comment]
@joshpayne I think we're fine with what we have now.

**Pull request merged**